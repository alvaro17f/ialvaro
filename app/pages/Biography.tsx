import Content from "@/components/Content";

export default function Biography() {
	return (
		<section id="biography">
			<Content>
				<div className="grid md:gap-10 md:grid-cols-[1fr_2fr] place-items-center h-[100dvh] mb-32 md:mb-0">
					<div className="mb-5">
						<a
							href="https://linkedin.com/in/alvarogarciamacias"
							rel="noreferrer"
							target="_blank"
						>
							<img
								src="/images/profile.png"
								alt="profile picture"
								width={200}
								height={200}
								className="border-[7px] rounded-full border-alvaro-white hover:border-dashed cursor-pointer"
							/>
						</a>
					</div>
					<div className="max-w-[75ch]">
						<h2 className="mb-4 text-4xl">
							Hello <span className="text-alvaro-danger">World</span>!
						</h2>
						<p>
							My name is{" "}
							<span className="text-xl text-alvaro-primary">√Ålvaro</span> and I
							am a Full Stack Developer.
							<br />
							I feel very fortunate to have found what I am passionate about and
							to make it my job.
							<br />
							I have also worked managing, training and leading marketing teams
							at a national level.
							<br />
							If there is one thing that characterises me, it is that I am 100%
							involved in everything I do.
							<br />
							I am hard-working, perfectionist and non-conformist.
							<br />
							I have a direct and creative working style. And I put all the
							energy I get from working in what I am so enthusiastic about.
						</p>
					</div>
				</div>
			</Content>
		</section>
	);
}
