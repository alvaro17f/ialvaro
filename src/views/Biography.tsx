import { Content } from "src/components/Content";
import data from "src/data/biography.json";
import { type ChangeEvent, useEffect, useState } from "react";

export default function Biography() {
	const [isMobile, setIsMobile] = useState(true);
	const [slider, setSlider] = useState(Math.round((data.length - 1) / 2));
	const [selectedBio, setSelectedBio] = useState(data[slider]);

	useEffect(() => {
		setIsMobile(window.innerWidth < 768);
	});

	useEffect(() => {
		if (slider < data.length) {
			setSelectedBio(data[slider]);
		}
	}, [slider]);

	const handleSlider = (e: ChangeEvent<HTMLInputElement>) => {
		setSlider(Number(e.target.value));
	};

	return (
		<section
			id="biography"
			className="md:h-[100dvh] grid items-center pt-24 md:pt-0"
		>
			<Content>
				<div className="grid grid-cols-1 md:gap-10 md:grid-cols-[1fr_2fr]">
					<div className="mb-5">
						<a
							href="https://linkedin.com/in/alvarogarciamacias"
							rel="noopener noreferrer"
							target="_blank"
						>
							<img
								src="/images/profile/profile.png"
								alt="profile"
								width={250}
								height={250}
								className="border-[7px] rounded-full border-alvaro-white hover:border-dashed cursor-pointer"
							/>
						</a>
					</div>
					<div className="max-w-[75ch] md:h-[300px] items-center md:text-justify grid  whitespace-pre-wrap">
						<div>
							<h1 className="mb-5 text-4xl">
								Hello <span className="text-alvaro-danger">World</span>!
							</h1>
							<p>
								My name is{" "}
								<span className="text-xl text-alvaro-primary">√Ålvaro</span>, and
								I am a Full Stack Developer.
							</p>
							{selectedBio.bio}
						</div>
					</div>
				</div>
				<div className="grid w-full grid-cols-2 mt-20 md:grid-cols-3 place-items-center">
					<button
						type="button"
						onClick={() => slider !== 0 && setSlider((slider) => slider - 1)}
						className="p-2 border text-alvaro-primary rounded-xl border-alvaro-primary md:hover:bg-alvaro-primary md:hover:text-alvaro-base"
					>
						Shortest
					</button>
					{!isMobile && (
						<input
							id="minmax-range"
							type="range"
							min="0"
							max={data.length - 1}
							value={slider}
							onChange={handleSlider}
							className="w-full h-2 bg-gray-200 rounded-lg outline-none appearance-none cursor-pointer dark:bg-gray-700"
						/>
					)}
					<button
						type="button"
						onClick={() =>
							slider !== data.length - 1 && setSlider((slider) => slider + 1)
						}
						className="p-2 border text-alvaro-danger rounded-xl border-alvaro-danger md:hover:bg-alvaro-danger md:hover:text-alvaro-base"
					>
						Longest
					</button>
				</div>
			</Content>
		</section>
	);
}
