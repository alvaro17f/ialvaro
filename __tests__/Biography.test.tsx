import Biography from "@/app/pages/Biography";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";

describe("<Biography />", () => {
	beforeEach(() => {
		const mockIntersectionObserver = vi.fn();
		mockIntersectionObserver.mockReturnValue({
			observe: () => null,
			unobserve: () => null,
			disconnect: () => null,
		});
		window.IntersectionObserver = mockIntersectionObserver;
		render(<Biography />);
	});
	afterEach(cleanup);

	it("should match the snapshot", () => {
		const { container } = render(<Biography />);
		expect(container).toMatchSnapshot();
	});

	it("should render an image", () => {
		expect(screen.getByRole("img")).toBeDefined();
	});

	it("should render a heading", () => {
		expect(
			screen.getByRole("heading", { level: 1, name: /hello world/i }),
		).toBeDefined();
	});

	it("should render a paragraph", () => {
		expect(screen.getByText(/Full Stack Developer/i)).toBeDefined();
	});

	it("should render a slider", () => {
		expect(screen.getByRole("slider")).toBeDefined();
	});

	it("should be able to change biography's value", () => {
		const slider = screen.getByRole("slider");
		expect(slider).toBeDefined();
		fireEvent.change(slider, { target: { value: 0 } });
	});
});
