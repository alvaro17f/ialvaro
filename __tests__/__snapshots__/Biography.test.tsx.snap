// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Biography /> > should match the snapshot 1`] = `
<div>
  <section
    class="md:h-[100dvh] grid items-center pt-24 md:pt-0"
    id="biography"
  >
    <section
      aria-label="content"
      class="p-5 mb-5 rounded-lg text-alvaro-white"
      style="opacity: 0;"
    >
      <article
        class="p-5"
      >
        <div
          class="grid grid-cols-1 md:gap-10 md:grid-cols-[1fr_2fr]"
        >
          <div
            class="mb-5"
          >
            <a
              href="https://linkedin.com/in/alvarogarciamacias"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="profile picture"
                class="border-[7px] rounded-full border-alvaro-white hover:border-dashed cursor-pointer"
                height="250"
                src="/images/profile/profile.png"
                width="250"
              />
            </a>
          </div>
          <div
            class="max-w-[75ch] md:h-[300px] items-center md:text-justify grid  whitespace-pre-wrap"
          >
            <div>
              <h1
                class="mb-5 text-4xl"
              >
                Hello 
                <span
                  class="text-alvaro-danger"
                >
                  World
                </span>
                !
              </h1>
              <p>
                My name is
                 
                <span
                  class="text-xl text-alvaro-primary"
                >
                  √Ålvaro
                </span>
                , and I am a Full Stack Developer.
              </p>
              I feel very fortunate to have found what I am passionate about and to make it my job.
            </div>
          </div>
        </div>
        <div
          class="grid w-full grid-cols-2 mt-20 md:grid-cols-3 place-items-center"
        >
          <button
            class="p-2 border text-alvaro-primary rounded-xl border-alvaro-primary md:hover:bg-alvaro-primary md:hover:text-alvaro-base"
          >
            Shortest
          </button>
          <input
            class="w-full h-2 bg-gray-200 rounded-lg outline-none appearance-none cursor-pointer dark:bg-gray-700"
            id="minmax-range"
            max="2"
            min="0"
            type="range"
            value="1"
          />
          <button
            class="p-2 border text-alvaro-danger rounded-xl border-alvaro-danger md:hover:bg-alvaro-danger md:hover:text-alvaro-base"
          >
            Longest
          </button>
        </div>
      </article>
    </section>
  </section>
</div>
`;
